crates/starknet-types-core/src/curve/affine_point.rs

```diff
@@ -44,12 +44,12 @@ impl AffinePoint {
 
     /// Returns the `x` coordinate of the point.
     pub fn x(&self) -> Felt {
-        Felt(*self.0.x())
+        Felt(*self.0.to_affine().x())
     }
 
     /// Returns the `y` coordinate of the point.
     pub fn y(&self) -> Felt {
-        Felt(*self.0.y())
+        Felt(*self.0.to_affine().y())
     }
 
```

Comment:
Don't you want to add a z() method too?

Reply:
I don't think it's necessary. The type is AffinePoint, I don't see why it should have a `z`.
If you think it make sense to also add `z`, I could add it. Or maybe a more general method `to_projective` that returns `(x,y,z)` projective coordinate